### PR TITLE
 Adding abides by interpretation tests

### PR DESCRIPTION
**Pull request by biachiarelli**

### CODE REVIEW

File: goald/quality/pragmatic/model/task.py [hunk 1]
```diff
@@ -60,26 +60,28 @@ def abidesByInterpretation(self, interp, current):
...
 
         if interp.getQualityConstraints([None]):
             for qc in interp.getQualityConstraints([None]):
-                feasible = self.checkQualityConstraint(qc, current)
+                try:
+                    myQC = self.myProvidedQuality(qc.metric, current)
+                    if myQC is not None:
+                        if not qc.abidesByQC(myQC, qc.metric):
+                            feasible = False
+                except:
```
Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

**Comment by gabrielsr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 4
           

Complexity increasing per file
==============================
- tests/test_data/mpers_metric.py  1
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  2
- goald/quality/pragmatic/model/task.py  2
         

Complexity decreasing per file
==============================
+ tests/quality_pragmaitc/test_task.py  -1
         

Clones added
============
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  9
- tests/quality_pragmaitc/test_task.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665&bid=16508323)

File: tests/quality_pragmaitc/test_experiment_scenarios_parts.py [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+from tests.test_data.mpers_metric import MpersMetrics
+
+import pytest
+
+
+# Contexts
+c1 = Context("c1")
+c2 = Context("c2")
+c3 = Context("c3")
+c4 = Context("c4")
```
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios_parts.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: tests/quality_pragmaitc/test_experiment_scenarios_parts.py [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+
+    identifyLocationByVoiceCallTask.addApplicableContext(c2)
+
+    accessLocationFromTriangulationTask.addApplicableContext(c2)
+
+    accessLocationFromGPSTask.addApplicableContext(c5)
+
+    # Goal Interpretation
+
+    qc1 = QualityConstraint(None, MpersMetrics.SECONDS, 180,
```
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios_parts.py]: Codacy Issue found: [function already defined line 352](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: goald/quality/pragmatic/model/task.py (hunk 1 above)
Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

**Comment by gabrielsr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- tests/quality_pragmaitc/test_experiment_scenarios.py  7
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  2
- tests/test_data/mpers_metric.py  1
- tests/test_data/mpers_model.py  1
- goald/quality/pragmatic/model/task.py  2
         

Complexity decreasing per file
==============================
+ tests/quality_pragmaitc/test_task.py  -1
         

Clones added
============
- tests/quality_pragmaitc/test_task.py  3
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  12
- tests/test_data/mpers_model.py  2
- tests/quality_pragmaitc/test_contexts_scenarios.py  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665&bid=16508323)

File: goald/quality/pragmatic/model/task.py (hunk 1 above)
Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: tests/quality_pragmaitc/test_experiment_scenarios_parts.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    plan = mpers.goals.centralReceivesInfoGoal.isAchievable(fullContext, None)
+
+    assert assertPlan(
+        plan,
+        [mpers.tasks.considerLastKnownLocationTask,
+         mpers.tasks.accessDataFromDatabaseTask,
+         mpers.tasks.sendInfoByInternetTask])
+
+
+def test_ContextSet1_medicalCareReachesGoal(mpers):
```
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios_parts.py]: Codacy Issue found: [function already defined line 74](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: tests/quality_pragmaitc/test_experiment_scenarios.py [hunk 5]
```diff
@@ -369,55 +48,158 @@ def test_MPEARS(rootGoal):
...
+                   mpers.contexts.c6,
+                   mpers.contexts.c7]
+
+    plan = mpers.rootGoal.isAchievable(fullContext, None)
+
+    assert plan is not None
+
+    for task in plan.getTasks():
+        found = 0
+
```
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

**Comment by gabrielsr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 4
           

Complexity increasing per file
==============================
- tests/test_data/mpers_model.py  1
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  2
- tests/test_data/mpers_metric.py  1
- tests/quality_pragmaitc/test_experiment_scenarios.py  7
- goald/quality/pragmatic/model/task.py  2
         

Complexity decreasing per file
==============================
+ tests/quality_pragmaitc/test_task.py  -1
         

Clones added
============
- tests/test_data/mpers_model.py  2
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  12
- tests/quality_pragmaitc/test_contexts_scenarios.py  9
- tests/quality_pragmaitc/test_task.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665&bid=16508323)

File: goald/quality/pragmatic/model/task.py (hunk 1 above)
Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: tests/quality_pragmaitc/test_experiment_scenarios_parts.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
```
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios_parts.py]: Codacy Issue found: [function already defined line 74](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: tests/quality_pragmaitc/test_experiment_scenarios.py [hunk 7]
```diff
@@ -369,55 +48,159 @@ def test_MPEARS(rootGoal):
...
```
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

**Comment by gabrielsr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 3
           

Complexity increasing per file
==============================
- tests/quality_pragmaitc/test_experiment_scenarios.py  7
- goald/quality/pragmatic/model/task.py  2
- tests/test_data/mpers_metric.py  1
- tests/test_data/mpers_model.py  1
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  2
         

Complexity decreasing per file
==============================
+ tests/quality_pragmaitc/test_task.py  -1
         

Clones added
============
- tests/quality_pragmaitc/test_task.py  3
- tests/test_data/mpers_model.py  2
- tests/quality_pragmaitc/test_experiment_scenarios_parts.py  12
         

Clones removed
==============
+ tests/quality_pragmaitc/test_contexts_scenarios.py  -11
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665&bid=16508323)

File: goald/quality/pragmatic/model/task.py (hunk 1 above)
Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py:80]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: tests/quality_pragmaitc/test_experiment_scenarios.py (hunk 7 above)
Comment (gabrielsr) [inline tests/quality_pragmaitc/test_experiment_scenarios.py:198]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

File: goald/quality/pragmatic/model/task.py (hunk 1 above)
Comment (gabrielsr) [inline goald/quality/pragmatic/model/task.py:80]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=5003665)

**Comment by rezendehugo** [comment]
> The name chosen for this variable is misleading at this point.
  adjusted

**Comment by rezendehugo** [comment]
> add None to the fullcontext makes sense?
  Fixed

**Pull request closed without merging**